src/flamenco/runtime/context/fd_exec_epoch_ctx.h

```diff
@@ -9,10 +9,10 @@
 /* fd_exec_epoch_ctx_t is the context that stays constant throughout
    an entire epoch. */
 
-struct __attribute__((aligned(8UL))) fd_exec_epoch_ctx {
-  ulong magic; /* ==FD_EXEC_EPOCH_CTX_MAGIC */
-  fd_features_t        features;
-  ulong epoch_bank_off;
+struct fd_exec_epoch_ctx_layout {
```

Comment:
Just so I understand, why not just use regular typed pointers? I think the layout macros should work with these.

Reply:
Ah they are offsets thanks!